homeassistant/components/bouncie/coordinator.py

```diff
@@ -0,0 +1,61 @@
+"""Bouncie data update class."""
+import datetime
+import logging
+
+from aiohttp.client_exceptions import ClientConnectorError
+from bounciepy import AsyncRESTAPIClient
+from bounciepy.exceptions import BouncieException, UnauthorizedError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_CLIENT_ID, CONF_CLIENT_SECRET
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import CONF_CODE, CONF_REDIRECT_URI, DOMAIN
+
+
+class BouncieDataUpdateCoordinator(DataUpdateCoordinator):
+    """Define Bouncie data updater."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        logger: logging.Logger,
+        config_entry: ConfigEntry,
+        update_interval: datetime.timedelta,
+    ) -> None:
+        """Init the coordinator."""
+        self._client_id = (config_entry.data[CONF_CLIENT_ID],)
+        self._client_secret = (config_entry.data[CONF_CLIENT_SECRET],)
+        self._redirect_uri = (config_entry.data[CONF_REDIRECT_URI],)
+        self._code = (config_entry.data[CONF_CODE],)
```

Comment:
`_client_secret`, `_redirect_uri` and `_code` aren't used anywhere else, and I think we should not expose those.

do not create var for them and use the config_entry directly on `AsyncRESTAPIClient`